src/SQL/Builder/DefaultUnionSQLBuilder.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\SQL\Builder;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Query\UnionQuery;
+
+use function count;
+use function implode;
+
+final class DefaultUnionSQLBuilder implements UnionSQLBuilder
+{
+    public function __construct(
+        private readonly AbstractPlatform $platform,
+    ) {
+    }
+
+    public function buildSQL(UnionQuery $query): string
+    {
+        $parts      = [];
+        $modifier   = $query->isDistinct() ? ' UNION ' : ' UNION ALL ';
```

Comment:
Do we really want to reuse the "distinct" flag to indicate a `UNION ALL`?

The QB's methods are usually named after the SQL syntax elements they represent in order to make them discoverable to developers. That's not really the case here.

Reply:
> Or should we use dedicated methods and not having a flag method ?

Sounds good.

Reply:
Changed to `union()/addUnion()/unionAll()/addUnionAll()` way not
reusing the `distinct()` flag. Thanks for the hint, this makes
it even clearer to use it and feels better.